hack/post-upgrade-checks.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -euo pipefail
+
+help="post-upgrade-checks.sh is used to perform tests after upgrade from the previous release.
+
+Usage:
+  post-upgrade-checks.sh [TEST_CLUSTER_CATALOG_NAME] [TEST_CLUSTER_EXTENSION_NAME]
+"
+
+if [[ "$#" -ne 2 ]]; then
+  echo "Illegal number of arguments passed"
+  echo "${help}"
+  exit 1
+fi
+
+TEST_CLUSTER_CATALOG_NAME=$1
+TEST_CLUSTER_EXTENSION_NAME=$2
+
+kubectl wait --for=condition=Available --timeout=60s -n olmv1-system deployment --all
+kubectl wait --for=condition=Unpacked --timeout=60s ClusterCatalog $TEST_CLUSTER_CATALOG_NAME
+kubectl wait --for=condition=Installed --timeout=60s ClusterExtension $TEST_CLUSTER_EXTENSION_NAME
```

Comment:
+1 on checking that the existing ClusterCatalog and ClusterExtension remain Unpacked and Installed. But two questions:
1. My instinct is that these waits might not actually check anything. We've already waited for them to be Unpacked and Installed prior to the upgrade, so they will still be Unpacked and Installed after the upgrade _at least_ until catalogd and operator-controller have reconciled them. Seems like we have a race condition here where these commands will execute before the upgraded reconcilers have finished reconciling them.
2. I think we should probably do a few extra tasks post-upgrade
   a. Change the ClusterExtension to specify a broader version range so that it finds and upgrades to a new version from the existing catalog
   b. Add a new ClusterExtension that installs another package from the existing ClusterCatalog


Also, potentially out-of-scope in this PR, but in the catalogd-specific upgrade tests, I think we'll want a scenario where the catalog image referenced by a ClusterCatalog receives an update after the upgrade, which would ensure that our polling logic doesn't break. That scenario may be relevant in operator-controller because we expect catalog updates to trigger reconciles (and potentially upgrade) ClusterExtension objects.

Reply:
> Are you suggesting to use "a" to trigger reconciliation? A workaround to my above question?

Note to self: `lastTransitionTime` won't change even if we bump a version. But we can check `observedGeneration`.

Reply:
I updated the scripts to initially install version `1.0.0` and then upgrade to `1.0.1` & added a check to make sure that the reconciliation actually works by checking installed contain again & its `observedGeneration`.

Reply:
I do think it would be better if we could
1. Wait for reconcile of the pre-upgrade state of the ClusterExtension before we modify it.
2. Make some assertions
3. Change the ClusterExtension
4. Make more assertions

To do (1), we can check `operator-controller` logs, which should have enough info to key off of.

Reply:
@joelanford could you please take a another look? Should be nearly there.